### ISSUE TITLE
nodejs xmldom svg support

### ISSUE THREAD

**Issue by psuedoSudo** [lindell/JsBarcode#149]
I'd like to create an svg Element in nodejs using [xmldom](https://github.com/jindw/xmldom) and then generate an svg barcode just like you can when using jsBarcode in a browser context. 

```js
const { DOMImplementation } = require('xmldom');
const document = new DOMImplementation().createDocument('http://www.w3.org/1999/xhtml', 'html', null);

const svgNode = document.createElementNS('http://www.w3.org/2000/svg', 'svg');

jsBarcode(svgNode, 'test', {
  format: 'CODE128C',
  displayValue: false,
  width: 4,
  height: 100
});
```

I'd hope it should be able to re-use the same `renderers.SVGRenderer` from [here](https://github.com/lindell/JsBarcode/blob/master/src/help/getRenderProperties.js#L39), would need to figure out how to tell if it's passed an xmldom svg element

**Comment by lindell** [comment]
You would probably need to do two things to make this work. First, as you say, you need to make the ´getRenderProperties´ to also detect other xml implementations (other than the browser one).

Then you need to pass the document down to the renderer and do something like
````
this.document = options.document || document;
````
In the [svgRenderer constructor](https://github.com/lindell/JsBarcode/blob/master/src/renderers/svg.js#L7) and replace all `document.create...` with `this.document.create...`

**Comment by lindell** [comment]
I assume that would work. If you make it work I will gladly accept that PR :)

**Comment by lindell** [comment]
I just pushed a [new branch](https://github.com/lindell/JsBarcode/tree/external-xml-impl) that should do this.

The problem is that there is no way to messure the size of the text which would make barcodes with very large font sizes render incorrectly. But otherwise it seems to work.

**Comment by lindell** [comment]
Works like this:

````javascript
    const svgNode = document.createElementNS('http://www.w3.org/2000/svg', 'svg');

    JsBarcode(svgNode, 'test', {
      xmlDocument: document
    });

    let xml = xmlSerializer.serializeToString(svgNode);
````

**Comment by psuedoSudo** [comment]
That was a crazy fast response time, looks like exactly what I need 😃. I'll give it a whirl and see

**Comment by psuedoSudo** [comment]
Worked like a charm!

**Comment by lindell** [comment]
Perfekt, I'm closing the issue then :)

**Comment by psuedoSudo** [comment]
Would it be possible to get this included/merged into a released version? should I open up a PR for that branch?

**Comment by lindell** [comment]
It is now merged in. But I can't unfortunately release from this computer (at work). I will do it the first thing when I come home.

**Comment by lindell** [comment]
Just released [JsBarcode v3.7.0](https://github.com/lindell/JsBarcode/releases/tag/v3.7.0) 😄

**Comment by psuedoSudo** [comment]
Thankyou!
